tests/test_requests.py

```diff
@@ -531,8 +531,7 @@ def test_basicauth_encodes_byte_strings(self):
             ('http://localhost:1', ConnectionError),
             # Inputing a URL that cannot be parsed should raise an InvalidURL error
             ('http://fe80::5054:ff:fe5a:fc0', InvalidURL),
-            # Inputing a URL that cannot be parsed by urllib3 should raise a ConnectionError, as it closes pool
-            ('http://1234567890123456789012345678901234567890123456789012345678901234.com', ConnectionError)
+            ('http://1234567890123456789012345678901234567890123456789012345678901234.com', InvalidURL)
```

Comment:
Behaviour is now consistent with the "IPv6" URL: an `InvalidURL` is raised, with an urllib3 `LocationParseError` in the stack trace.